NewsFeed/Common/Views/ScrollableSegmentedControl.swift

```diff
@@ -0,0 +1,235 @@
+            sendActions(for: .valueChanged)
+            previousButton?.isSelected = false
+            UIButton.animate(withDuration: 0.3, animations: {
+                previousButton?.transform = CGAffineTransform.identity
+            })
+        }
+ 
+        sender.isSelected = true
+    }
+    
+// Private
+    // Create buttons
+    private func createButtons() {
+        for button in buttonsArray {
+            button.removeFromSuperview()
+        }
+        buttonsArray.removeAll()
+        
+        var buttonTag: Int = 0
+        
+        for title in buttonTitles {
+            let button = UIButton(type: .custom)
+        
+            button.setTitleColor(UIColor.lightGray, for: .normal)
+            button.setTitleColor(UIColor.black, for: .selected)
+            button.addTarget(self, action: #selector(buttonTapped), for: .touchUpInside)
+            button.setTitle(title, for: .normal)
+            
+            button.tag = buttonTag
+            
+            buttonTag += 1
+            
+            buttonsArray.append(button)
+        }
+    }
+    
+    private func updateUI() {
+        var numberOfPaddings: CGFloat = -1
+        var buttonX: CGFloat = buttonMargin
+        var contentSizeWidth: CGFloat = 0
+        
+        // Set button frame
+        for button in buttonsArray {
+            button.transform = CGAffineTransform.identity
+            button.titleLabel?.font = font
+            
+            let buttonSize = calculateButtonSize(title: (button.titleLabel?.text)!, fontAttributes: [NSAttributedStringKey.font: font!])
+            
+            button.frame = CGRect(x: buttonX,
+                                  y: (frame.size.height - buttonSize.height) / 2,
+                                  width: buttonSize.width,
+                                  height: buttonSize.height)
+            
+            buttonX += button.frame.size.width + buttonPadding
+            
+            scrollView.addSubview(button)
+            
+            numberOfPaddings += 1
+            
+            contentSizeWidth += button.frame.size.width
+        }
+        
+        scrollView.contentSize.width = contentSizeWidth + (buttonPadding * (CGFloat(buttonsArray.count) - 1)) + (2 * buttonMargin)
+    }
+    
+    func calculateButtonSize(title: String, fontAttributes: [NSAttributedStringKey : Any]) -> CGSize {
+        let buttonSize = CGSize(width: (title.size(withAttributes: fontAttributes).width),
+                                height: (title.size(withAttributes: fontAttributes).height))
+        return buttonSize
+    }
+    
+    // Draw separator line
+    private func drawSeparatorLine() {
+        let separatorHeight: CGFloat = 1
+        let separatorWidth: CGFloat
+        
+        if scrollView.contentSize.width <= frame.size.width {
+            separatorWidth = frame.size.width
+        } else {
+            separatorWidth = scrollView.contentSize.width
+        }
```

Comment:
Separator width can simply be `frame.size.width`, since you are adding it to `self`, not the scroll view.